linkml_runtime/utils/schemaview.py

```diff
@@ -108,25 +108,6 @@ def is_absolute_path(path: str) -> bool:
     drive, tail = os.path.splitdrive(norm_path)
     return bool(drive and tail)
 
-def _resolve_import(source_sch: str, imported_sch: str) -> str:
-    if os.path.isabs(imported_sch):
-        # Absolute import paths are not modified
-        return imported_sch
-    if urlparse(imported_sch).scheme:
```

Comment:
`urlparse` does not check for URL validity, which isn't obvious if you haven't read the docs for the function. A better test here would be for `:`, which should only occur in well-formed URLs and CURIEs, but not in file system paths.

Reply:
Also -- `imported_sch` starting with `file://` and `file:///` do not resolve and `http` / `https` / etc. will be rejected unless there's a mapping in the `prefixes` section of the schema.

Reply:
plz add failing tests illustrating these cases and then fix :)

Reply:
It isn't clear to me what the desired behaviour is here, which is the main reason why I didn't do anything. I thought that the point of the `prefixes` section was so that external resources could be referred to without using URLs - specify the URL there and the importer Does The Right Thing automatically. The docs/code are a bit ambiguous as the code enforces prefixes but the docs suggest URLs are OK.

`file://` (or `file:///`) don't work at all due to bugs in the underlying `hbreader` package. I don't know whether this is a problem throughout the codebase, but I think I would consider ditching the package and using something more standard if so.

Reply:
ETA: I've just realised it's a code comment that's leading me astray:
```
                    # origin schema. Imports can be a URI or Curie, and imports from the same
                    # directory don't require a ./, so if the current (sn) import is a relative
```
 The docs are pretty clear that the field contents should be local paths or CURIEs.

Reply:
I would assume the metamodel spec of xsd:anyURI is correct over the docs, since the metamodel is a formal specification? what makes you say the docs are clear(er than the metamodel?)

Reply:
The section below describes how to import external schemas (e.g. from the web) by assigning them a prefix, giving the example of the `linkml:types` schema.

Reply:
right, but in general i would think that a formal specification would be the source of truth, and docs are just a description of it that can be imprecise. for example it's not really clear what a "local import" is. Since i'm a language user i interpret that as meaning "local path," but the spec has no such ambiguity. the same thing that allows the prefix notation is what would allow URIs - the range is `URIOrCurie` and those curie prefixes expand into URIs.

Reply:
Well that's certainly a bug

Reply:
I think we need a core LinkML person to chime in here.

Reply:
I personally would say that is out of scope for this PR, i don't think you need to fix all of URI resolution rn. i think the shortcomings of the hbreader are known, and also think that the only way could feasibly resolve the problem is make a plugin architecture that allowed people to handle all the various schemes that one might want to use to resolve a URI.

Reply:
Yeah, it did seem like a bit too much for this PR -- I wasn't sure from your comment whether or not you thought it needed fixing now or whether it was something that could be added to the list.